### PR TITLE
Use the `eager` Pkg server registry, by default

### LINKED ISSUE #14: Reconsider using Pkg server by default for the registry
In 1.7, Pkg has moved to use a compressed registry by default. This avoids unpacking ~22 000 files and has some good performance improvements for the first time installing a registry (which is what CI does all the time):

**git** :

```jl
julia> @time Pkg.add("Example")
  Installing known registries into `~/.julia`
     Cloning registry from "https://github.com/JuliaRegistries/General.git"
       Added registry `General` to `~/.julia/registries/General`
   Resolving package versions...
    Updating `/tmp/jl_nJuAcb/Project.toml`
  [7876af07] + Example v0.5.3
    Updating `/tmp/jl_nJuAcb/Manifest.toml`
  [7876af07] + Example v0.5.3
 15.838710 seconds (8.68 M allocations: 301.542 MiB, 0.29% gc time, 1.04% compilation time)
```

**pkg server**:

```
julia> @time Pkg.add("Example")
  Installing known registries into `~/.julia`
    Updating registry at `~/.julia/registries/General.toml`
   Resolving package versions...
    Updating `/tmp/jl_16gHZs/Project.toml`
  [7876af07] + Example v0.5.3
    Updating `/tmp/jl_16gHZs/Manifest.toml`
  [7876af07] + Example v0.5.3
  2.753616 seconds (5.57 M allocations: 355.760 MiB, 1.80% gc time, 0.09% compilation time)
```

This is on my Linux machine with a fast hard drive. On windows, the difference is much larger.

By the time https://github.com/julia-actions/julia-buildpkg/pull/6 was merged, the Pkg server would only serve the updated General once all the packages and artifacts were available. This is no longer the case and the Pkg server serves the updated registry straight away.

### PR DESCRIPTION
**Pull request by DilumAluthge**
Fixes #14

### CODE REVIEW

**Comment by fredrikekre** [comment]
What is needed to push this over the finish line? For example for Literate.jl, CI takes 7m14s for Windows but 1m45s for Ubuntu. It seems pretty ridiculous that the Julia ecosystem waste so much CI for this.

I am sure eager registry is fine, but an alternative would be to download the current snapshot of General#master from GitHub directly.

File: action.yml [hunk 1]
```diff
@@ -13,22 +13,47 @@ inputs:
...
-    shell: bash
-
-  - run: julia --color=yes --project=${{ inputs.project }} -e 'using Pkg; if VERSION >= v"1.1.0-rc1"; Pkg.build(verbose=true); else Pkg.build(); end'
-    shell: bash
+  - run: |
+      import Pkg
+      pref_key = "JULIA_PKG_SERVER_REGISTRY_PREFERENCE"
+      pref_value = strip(get(ENV, pref_key, "eager"))
+      ENV[pref_key] = pref_value
+      input = strip(ENV["PROPAGATE_INPUT"])
```
Comment (KristofferC) [inline action.yml:28]: What is this `PROPAGATE_INPUT` stuff? I'm confused why this fix is not a lot simpler; just set the env variable and done? It seems it is adding these extra options that are pretty unlikely to be used in practice. If someone wants some special behavior they can just add that to their own CI files.

Comment (fredrikekre) [reply action.yml:28]: I was about to say the same. If you need this then it should go in the root environment key of the CI job description. It seems strange that it is "hidden" in the buildpkg action (which most packages should not need to use anyway).

**Comment by SaschaMann** [comment]
superseded by https://github.com/julia-actions/julia-buildpkg/pull/31 but thanks for opening this PR!

**Pull request closed without merging**